Mathlib/Algebra/Group/Semiconj/Basic.lean

```diff
@@ -63,5 +63,9 @@ variable [Group G] {a x y : G}
 #align semiconj_by.zpow_right SemiconjBy.zpow_right
 #align add_semiconj_by.zsmul_right AddSemiconjBy.zsmul_right
 
+variable (a) in
+@[to_additive] lemma eq_one_iff (h : SemiconjBy a x y): x = 1 ↔ y = 1 := by
```

Comment:
Why do you want `a` to be an explicit variable here?

Reply:
because you can prevent the occurance of a data-valued goal, in the case of using this lemma with `rw` without `h`. It still leaves `y` unspecified, but this is at least a partial fix for that problem.

Reply:
if you know of a better formulation to allow users to be able to prevent data-valued goals when using this lemma, please do suggest it.

Reply:
E.g., Lean 4 allows `rw [SemiconjBy.eq_one_iff (a := a) (y := y)]`. If `SemiconjBy` is not available in the environment, then I think that a better style is to add `have H : SemiconjBy a x y` first.

Reply:
Anyway, it's a minor issue. You can choose either option, then merge. Thanks!
bors d+